src/plugins/gds_mt/gds_mt_utils.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Serapheim Dimitropoulos, WekaIO Ltd.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include "common/nixl_log.h"
+#include "gds_mt_utils.h"
+
+nixl_status_t gdsMtUtil::registerFileHandle(int fd,
+                                            size_t size,
+                                            std::string metaInfo,
+                                            gdsMtFileHandle& gdsMtHandle)
+{
+    CUfileError_t status;
+    CUfileDescr_t descr;
+    CUfileHandle_t handle;
+
+    descr = {};
+    descr.handle.fd = fd;
+    descr.type = CU_FILE_HANDLE_TYPE_OPAQUE_FD;
+
+    status = cuFileHandleRegister(&handle, &descr);
+    if (status.err != CU_FILE_SUCCESS) {
+        NIXL_ERROR << "GDS_MT: file register error: error=" << status.err << ", fd=" << fd;
+        return NIXL_ERR_BACKEND;
+    }
+
+    gdsMtHandle.cu_fhandle = handle;
+    gdsMtHandle.fd = fd;
+    gdsMtHandle.size = size;
+    gdsMtHandle.metadata = metaInfo;
+
+    return NIXL_SUCCESS;
+}
+
+nixl_status_t gdsMtUtil::registerBufHandle(void *ptr,
+                                           size_t size,
+                                           int flags)
+{
+    CUfileError_t status;
+
+    status = cuFileBufRegister(ptr, size, flags);
+    if (status.err != CU_FILE_SUCCESS) {
+        NIXL_ERROR << "GDS_MT: warning: buffer registration failed - will use compat mode: error=" << status.err;
```

Comment:
either write an error and return a status other than NIXL_SUCCESS, or print a warning and return NIXL_SUCCESS.